backend/src/libs/packages/token/libs/token.interface.ts

```diff
@@ -1,6 +1,8 @@
+import { type JWTPayload } from 'jose';
+
 interface IToken {
   create(payload: unknown): Promise<string>;
-  verify(): void;
-  decode(): void;
+  verify(token: string): Promise<JWTPayload>;
```

Comment:
```suggestion
  verify<T>(token: string): Promise<JWTPayload & T>;
```
should be generic. we need to provide all properties that we add to jwt payload